apps/govquests-frontend/src/domains/notifications/components/NotificationToaster.tsx

```diff
@@ -0,0 +1,45 @@
+import { Toaster } from "@/components/ui/toaster";
+import { useToast } from "@/hooks/use-toast";
+import { useEffect } from "react";
+import { useNotificationDiff } from "../hooks/useNotificationDiff";
+import { useNotifications } from "../hooks/useNotifications";
+import { NotificationStrategyFactory } from "../strategies/NotificationStrategyFactory";
+import { NotificationNode } from "../types/notificationTypes";
+
+const NotificationToaster = () => {
+  const { data } = useNotifications();
+  const { toast } = useToast();
+
+  const notifications = data?.pages?.flatMap(
+    (value) => value.notifications.edges,
+  );
+  const { getNewNotifications } = useNotificationDiff(notifications);
+
+  useEffect(() => {
+    const newNotifications = getNewNotifications();
+
+    newNotifications.forEach((notification) => {
+      displayNotification(notification.node, toast);
+    });
+  }, [notifications]);
+
+  return <Toaster />;
```

Comment:
you can return the toaster to be used by anyone in the main layout, and just call this hook to set up the notification handling part

Reply:
Makes sense to me, do you think I should call this hook on NotificationBell component? Or Header, for example?

Reply:
up to you =) I'd call it in the base layout I think